### PR TITLE
Change the display of events from cards to lists

### LINKED ISSUE #505: Change the display of events from cards to lists

### PR DESCRIPTION
**Pull request by yuqi88**
## Description

- Change the display of events from cards to lists, and styled.
- The order of each section is ordered by the start_date of the events
- Only the grade for coming-soon event displays TBA
- For the teachers, change the wording of the link button
- For the students, the link to the Events page is enabled only when the challenge is available

## Types of changes

What types of changes does your code introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (only improve the code quality no change in functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] Issue has been created
- [ ] Every file touched is linted
- [ ] Every file touched has tests that cover all the funcitonality with green coverage
- [ ] Documentation is updated (if applicable).

## Issue
closes #505 

## Tests

--

## Screenshots (if appropriate):
(1)
image

(2)After clicking the orange Assignment button:
image
Clicking the non-coloured Assignment button displays (1)

(3)After clicking the orange Exam button from (1):
image
Clicking the non-coloured Exam button displays (1)

(4)After clicking both the orange Assignment button & orange Exam button, the page displays (1)

### CODE REVIEW

**Comment by yuqi88** [comment]
update:
- change the filter to a dropdown
- improve the filter functions
- add the import event card back
- style the edit event button in the list
- add responsiveness

filter: view all assignments & exams
image

filter: view assignments only
image

filter: view exams only
image

**Comment by yuqi88** [comment]
Responsiveness:
image
image

File: src/app/course/course-events-snippet/course-events-snippet.component.html [hunk 1]
```diff
@@ -1,100 +1,177 @@
...
-        Cancel
-    </button>
-</ng-template>
+        </ng-container>
+        <ng-template
+            #importDialog
+            let-observer
+        >
+            <ng-container *ngIf="courseEvents">
+                <div class="import-events tui-space_top-8">
```
Comment (yuqi88) [inline src/app/course/course-events-snippet/course-events-snippet.component.html:33]: Not sure what this does. Is it okay if this stays here?

**Comment by yuqi88** [comment]
alignment:
image

image

image

**Review by keyvankhademi — CHANGES_REQUESTED** [review]
Please fix the lint issues. You can also modify eslint to add tabs to the switch case.

**Review by keyvankhademi — CHANGES_REQUESTED** [review]
Generally looks good.
Can we please make the add event a button instead of that big island?
Also please make sure the tests are passing.
and pull from master and resolve the conflicts.

**Comment by yuqi88** [comment]
image

**Pull request merged**